engine/src/main/java/org/terasology/entitySystem/systems/RenderSystem.java

```diff
@@ -16,16 +16,32 @@
 package org.terasology.entitySystem.systems;
 
 /**
+ * Interface for component systems that needs to be updated every time certain game elements are rendered (generally to render something).
```

Comment:
Tricky tricky tricky. I'm struggling to find a proper description for this.

I guess we could write:

_Instances implementing this interface see their methods called at specific stages during the rendering process, with an OpenGL state compatible with what needs to be rendered - i.e. opaque objects (a fairly standard opengl state) vs semi-transparent objects (opengl has blending enabled). This allows a RenderSystem to update its own entities or inject its own objects for rendering._  

That been said, long term this interface might disappear.